targets/TARGET_STM/TARGET_STM32H7/STM32Cube_FW/stm32h7xx_hal_conf.h

```diff
@@ -215,8 +215,16 @@
 #define  USE_HAL_WWDG_REGISTER_CALLBACKS    0U /* WWDG register callback disabled    */
 
 /* ########################### Ethernet Configuration ######################### */
+#ifdef MBED_CONF_STM32_EMAC_ETH_RXBUFNB
+/* default value in features/netsocket/emac-drivers/TARGET_STM/mbed_lib.json */
 #define ETH_TX_DESC_CNT         MBED_CONF_STM32_EMAC_ETH_TXBUFNB  /* number of Ethernet Tx DMA descriptors */
 #define ETH_RX_DESC_CNT         MBED_CONF_STM32_EMAC_ETH_RXBUFNB  /* number of Ethernet Rx DMA descriptors */
+#else
+/* ex: bare metal profile */
+#define ETH_TX_DESC_CNT         0  /* Tx buffers of size ETH_TX_BUF_SIZE  */
+#define ETH_RX_DESC_CNT         0  /* Rx buffers of size ETH_RX_BUF_SIZE  */
```

Comment:
What does "0" mean?
Is Ethernet expect to work on Baremetal?

Reply:
I don't think so, Ethernet needs threads

Reply:
Ah! then it might be better to disable ethernet rather than silently compiling, right?
And perhaps if the network API is used in bare-metal, there should be an error (I'm not sure where exactly the check should be introduced but it might be a tools/build-system enhancement).
For reference: https://os.mbed.com/docs/mbed-os/v6.0/bare-metal/index.html

Reply:
Yes, Ethernet is disabled with baremetal, but Ethernet ST Cube drivers are compiled.
And these 2 macro ETH_RX/TX_DESC_CNT are used in ST files. Definition were defined at ST Cube level, but I moved them in emac directory for mbed. So we need to redefine it when emac is not present.

Reply:
@jeromecoutant might be useful to have the reason for having this set to 0 in the commit message - does that mean no allocated memory as they are 0 or symbols are protected somehow? I've already started CI ,can restart later

Reply:
As emac is disabled, you can set the value you want....